core/src/bundle/usBundleInfo.cpp

```diff
@@ -22,13 +22,25 @@
 
 
 #include "usBundleInfo.h"
+#include "usUtils_p.h"
 
+#include <stdlib.h>
 namespace us {
 
-BundleInfo::BundleInfo(const std::string& name)
-  : name(name)
-  , id(0)
+BundleInfo::BundleInfo(const std::string& location, const std::string& name)
+: name(name)
+, id(0)
 {
+#if _WIN32
+  this->location = location;
+#else
+  // resolve any symlinks. 
+  char filePath[PATH_MAX];
+  char* resultPath = ::realpath(location.c_str(), filePath);
+  if (resultPath) {
```

Comment:
If realpath fails, teh return value is NULL. It results in a BundleInfo with empty location and name. The Bundle creation fails with runtime error. Its consistent with what happens today when you send an invalid path to InstallBundle function.